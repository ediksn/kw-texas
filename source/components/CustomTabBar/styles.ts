import { StyleSheet } from 'react-native'
import { theme } from '~/constants'

export const styles = (focused: boolean) =>
  StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'flex-start'
    },
    selectedBar: {
      backgroundColor: focused ? theme.activeColor : theme.backgroundWhite,
      paddingTop: '5%',
      paddingLeft: '30%',
      paddingRight: '30%',
      borderBottomLeftRadius: 100,
      borderBottomRightRadius: 100,
      marginBottom: '10%'
    },
    icon: {
      color: focused ? theme.activeColor : theme.inActiveColor,
      marginTop: '5%',
      height: 23,
      width: 22.5,
      paddingLeft: '2%'
    },
    photo: {
      aspectRatio: 1,
      width: 30,
      borderRadius: 30,
      height: 30
    }
  })
