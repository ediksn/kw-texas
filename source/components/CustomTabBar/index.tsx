import React, { memo } from 'react'
import { View, Image, ImageStyle } from 'react-native'
import { createIconSetFromIcoMoon } from 'react-native-vector-icons'
import Icomoon from '~/selection.json'
import { styles } from './styles'

const Icon = createIconSetFromIcoMoon(Icomoon)

interface Props {
  focused: boolean
  name?: string
  PNG?: any
  imageURL?: string
  imageStyle?: ImageStyle
}
const CustomTabBar = ({ name, focused, PNG, imageURL, imageStyle }: Props) => {
  return (
    <View style={styles(focused).container}>
      <View style={styles(focused).selectedBar} />
      {name ? (
        <Icon name={name} size={23} style={styles(focused).icon} />
      ) : PNG ? (
        <Image style={[imageStyle]} resizeMode='contain' resizeMethod='auto' source={PNG} />
      ) : (
        <Image style={[imageStyle]} resizeMode='cover' resizeMethod='resize' source={{ uri: imageURL }} />
      )}
    </View>
  )
}

export default memo(CustomTabBar)
