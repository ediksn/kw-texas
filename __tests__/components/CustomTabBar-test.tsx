import React from 'react'
import renderer from 'react-test-renderer'
import HomeFilled from 'assets/images/home-filled.png'
import CustomTabBar from '~/components/CustomTabBar'

test('renders correctly', () => {
  const tabBar = renderer.create(<CustomTabBar PNG={HomeFilled} focused />).toJSON()
  expect(tabBar).toMatchSnapshot()
})
