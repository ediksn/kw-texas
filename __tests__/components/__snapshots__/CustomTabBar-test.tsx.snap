// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#006C84",
        "borderBottomLeftRadius": 100,
        "borderBottomRightRadius": 100,
        "marginBottom": 11.770588235294117,
        "paddingLeft": 45,
        "paddingRight": 41.19705882352941,
        "paddingTop": "5%",
      }
    }
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 21,
        },
        Object {
          "color": "#006C84",
          "marginTop": 5.885294117647058,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î¤„
  </Text>
</View>
`;
